ResearchKit/Common/ORKAxisView.m

```diff
@@ -0,0 +1,148 @@
+        [constraints addObject:[NSLayoutConstraint constraintWithItem:label
+                                                            attribute:NSLayoutAttributeCenterX
+                                                            relatedBy:NSLayoutRelationEqual
+                                                               toItem: label.superview
+                                                            attribute:NSLayoutAttributeLeading
+                                                           multiplier:1
+                                                             constant:offset]];
+        if (i == self.titleLabels.count - 1) {
+            [constraints addObject:[NSLayoutConstraint constraintWithItem:label
+                                                                attribute:NSLayoutAttributeHeight
+                                                                relatedBy:NSLayoutRelationEqual
+                                                                   toItem: label.superview
+                                                                attribute:NSLayoutAttributeHeight
+                                                               multiplier:1
+                                                                 constant: -_lastLabelPadding]];
+            [constraints addObject:[NSLayoutConstraint constraintWithItem:label
+                                                                attribute:NSLayoutAttributeWidth
+                                                                relatedBy:NSLayoutRelationEqual
+                                                                   toItem: label.superview
+                                                                attribute: NSLayoutAttributeHeight
+                                                               multiplier:1
+                                                                 constant: -_lastLabelPadding]];
+        }
+    }
+    [NSLayoutConstraint activateConstraints:constraints];
+    _constraints = constraints;
+}
+
+- (void)setupTitles:(NSArray *)titles {
+    
+    for (NSUInteger i = 0; i < titles.count; i++) {
+        
+        UILabel *label = [UILabel new];
+        label.text = titles[i];
+        label.font = [UIFont systemFontOfSize:12.0];
+        label.numberOfLines = 2;
+        label.textAlignment = NSTextAlignmentCenter;
+        label.adjustsFontSizeToFitWidth = YES;
+        label.minimumScaleFactor = 0.7;
+        label.textColor = self.tintColor;
+        label.translatesAutoresizingMaskIntoConstraints = NO;
+        
+        if (i == titles.count - 1) {
+            label.textColor = [UIColor whiteColor];
+            label.backgroundColor = self.tintColor;
+            label.layer.cornerRadius = (self.bounds.size.height - _lastLabelPadding) * 0.5;
+            label.layer.masksToBounds = YES;
+        }
+        
+        [self addSubview:label];
+        [self.titleLabels addObject:label];
+    }
```

Comment:
We probably need a `[self setNeedsUpdateConstraints]` call here, in case the label amount changes midway through.